lib/grpc/server/adapters/cowboy/handler.ex

```diff
@@ -136,11 +132,32 @@ defmodule GRPC.Server.Adapters.Cowboy.Handler do
           content_type
       end
 
-    {:ok, access_mode, subtype} = extract_subtype(content_type)
+    http_method = extract_http_method(req)
+
+    {:ok, access_mode, subtype} =
+      case extract_subtype(content_type) do
+        {:ok, :unknown, "unknown"} ->
+          if http_method == :post do
+            {:ok, :grpc, "proto"}
+          else
+            {:ok, :http_transcoding, "json"}
+          end
+
+        resp ->
+          resp
+      end
+
     access_mode = resolve_access_mode(req, access_mode, subtype)
     {:ok, access_mode, subtype, content_type}
   end
 
+  defp extract_http_method(req) do
+    req
+    |> :cowboy_req.method()
+    |> String.downcase()
+    |> String.to_existing_atom()
```

Comment:
I think we could get away with just the downcase string comparison to avoid unnecessary raises